new docs - settings, client-side forms, document preview (draft) (#21)
herflis/concepts-temp: content/trash.md

The size of the trash and the time frame while the contents can be recovered from the trash is set by the administrator. Trash feature can be switched on or off globally.

# Delete operation capabilities
Delete operation allows users to transfer content (documents, folders or whole workspaces) into the Trash Bin - or delete them permanently.

Contents can be deleted through API calls, but these actions are also available on the admin ui:

You can access delete action for basically every content in the doclib by clicking context menu (...) or right click on item.
![delete action](./img/delete_contextmenu.png)

By default configuration, a confirmation popup appears, displaying content name and path, with an additional option to permanently delete the content (skipping trash). 
![confirm delete](./img/confirm.png)

Deleted (trashed) items can be found in a specific folder having some extra parameter like retention time, quota for size and number of content. These limits can be customized.
![trash](./img/trasheditem.png)

> Actions can also be performed on a trashed item. It can be moved to a specific location (restore from trash), deleted, edited and so on.

Reviewer: Somewhere here I would mention that content from trash could be restored to a container chosen by the user. By defatult this container is the original parent of the deleted content, but the restore target could be chosen freely, keeping in mind that the type of the restored content should be allowed on the chosen container.

> Actions can also be performed on a trashed item. It can be moved to a specific location (restore from trash), deleted, edited and so on.

Author: ```suggestion
Content from trash could be restored to a container chosen by the user. By default this container is the original parent of the deleted content, but the restore target could be chosen freely,

**important:**  type of the restored content should be allowed on the chosen container.

Additional actions can also be performed on trashed items like edit or delete permanently.
```

### Enable/disable the trash feature on a container

You can enable or disable the trash feature on almost any container content. By default, the feature is enabled and deleted content go to the trash. You can change this on the Edit page of any folder or list. This setting effects only the container (list, workspace) you edit, and not the child folders.

# for devs

### trash bag

The Trash Bag is a simple container content that 'wraps' the deleted content. You can access the deleted item through the DeletedContent property of the Trash Bag.

```
var dc = trashBag.DeletedContent;
if (dc != null)
	contentLabel.Text = dc.DisplayName;
```

### permissions

The content moved to the trash will preserve its actual permissions and the container Trash Bag will get all its inherited permissions. This means that if you delete a content that only few users can see, other users won't see it in the trash either.

> 
> # Trash - for Developers
> # Trash - for System operators
> 
> https://wiki.sensenet.com/Trash
> 
> és itt a links részben található cuccok

Author: ```suggestion
```

metaDescription: "sensenet Trash"
---

# introduction
sensenet has a trash feature. When you delete a document or a folder, and the trash feature is enabled (enabled by default), the content will be wrapped into a special container called the Trash Bag, and will be moved to the Trash Bin (/Root/Trash). 

The Trash Bin is a central place for deleted documents where you can browse, restore or delete content permanently. It is also possible to provide a local (workspace-level) trash for users.

The size of the trash and the time frame while the contents can be recovered from the trash is set by the administrator. Trash feature can be switched on or off globally.

# Delete operation capabilities
Delete operation allows users to transfer content (documents, folders or whole workspaces) into the Trash Bin - or delete them permanently.

> - delete content that go to the trash
> - delete content permanently, without using trash
> - enable or disable trash locally or globally
> - enable or disable trash for a content type
> 

Author: ```suggestion
```

- Autofilters: disable
- Query: +InTree:'/Root/Trash' +TypeIs:TrashBag +WorkspaceId:@@CurrentWorkspace.Id@@

All content deleted under the workspace will be visible here and can be restored or deleted permanently. The deleted content are still accessible and restorable from the global Trash and all size and date settings are still applied.

# trash bin options
Trash Bin options can be edited through API. 

### Enable/disable the trash feature globally

If the trash is enabled in the system, deleted content can be moved to the Trash Bin. You can find them on the path /Root/Trash.
If you disable the trash, previously deleted content won't disappear, you or the owner can restore them freely. Content deleted with disabled trash will be removed permanently from the database.

### Set the retention time of deleted contents

The value set here is the days the content should stay in the trash before deleting it permanently. If the value is greater than 0, users (or any automatism) cannot remove the content from the trash before the expiration date. Changing this value does not effect previously deleted content.

### Set the size of the Trash Bin

> The size quota is the amount of content in megabytes that can be stored in the trash. This value is only a UI hint, it does not effect deleting content. If the size is exceeded the Trash Bin main page will display a message about how much space is used. The administrator should take care of purging content from the trash manually.

Author: this may be obsolete (from content explorer time).
Please confirm.
